project-koku/koku
# [COST-1847] unleash client for trino enablement

This PR adds the unleash client and allows us to utilize Unleash to enable trino processing for account/schema numbers. All the other mechanisms in place to enable trino processing will still work, so any `ENABLE_TRINO_SOURCES`, `ENABLE_TRINO_SOURCE_TYPE`, or `ENABLE_TRINO_ACCOUNTS` that are already defined will function the same way.

Adding the unleash client here allows us to add new accounts without the hassle of updating the deploy.yaml in app-interface.

Testing:
(this is easier to see _without_ trino running)
1. make remove-db
2. make docker-up-min
2. make create-test-customer
3. make load-test-customer
4. See the everything succeeds
5. Delete all sources and cost models from the db (yea, this is a pain)
6. Log into unleash (http://localhost:4242/) and add `acct10001` to the schema-name list in the schema-strategy for the `cost-trino-processor` feature flag
7. recreate the test customers and load in data
8. watch _everything_ fail to ingest (because all data is going thru trino and trino is not running)

[Smokes](https://ci.ext.devshift.net/job/project-koku-koku-pr-check/1617/testReport/):
Manually ran the 2 failures:
```
$ iqe tests plugin cost_management -k test_api_ocp_source_crud_upload_service  -m cost_smoke
1 passed, 6120 deselected, 5 warnings in 224.55s (0:03:44)
$ iqe tests plugin cost_management -k test_api_ocp_sources_shared_rates_recalculation -m cost_smoke
1 passed, 6120 deselected, 5 warnings in 31.30s
```
